### ISSUE TITLE
Give users an option to specify their home server

### ISSUE THREAD

**Issue by moddedBear** [tvler/streetpass#4]
Might be nice to give users the option to specify their home server. Then when clicking a profile from their StreetPass list they can be taken to that profile as viewed on their home server rather than the profile's server. This would make follows and interactions a bit easier.

Thanks for the extension!

**Comment by tvler** [comment]
this is a great idea. more UI/UX features after the safari launch!

**Comment by fauno** [comment]
This would be really helpful because we are developing ActivityPub-enabled sites and StreetPass is finding the profile but then it sends you to the homepage (since the homepage is the profile itself).  If it sends you to your own instance with a search on, it'll be really easy to follow!

**Comment by tvler** [comment]
This is in the roadmap! I needed to start saving usernames (not just profile urls) for this to be possible, which just handed in v2023.13! It would’ve been possible to infer usernames from a profile url, but that isn’t technically a part of the spec and I found enough edge cases to decide it needed to be done a little more formally. After that I can expose a settings screen to set the url scheme you want mastodon links to be. 

Plan is for v2023.14 I’m gonna ship an update to start showing avatar images in StreetPass, and get to this in 2023.15
